motor/providers/microsoft/provider.go

```diff
@@ -10,18 +10,26 @@ import (
 	"go.mondoo.com/cnquery/motor/vault"
 )
 
+type microsoftAssetType int32
+
+const (
+	OptionTenantID       = "tenant-id"
+	OptionClientID       = "client-id"
+	OptionDataReport     = "mondoo-ms365-datareport"
+	OptionSubscriptionID = "subscription-id"
+)
+
 const (
-	OptionTenantID   = "tenant-id"
-	OptionClientID   = "client-id"
-	OptionDataReport = "mondoo-ms365-datareport"
+	ms365 microsoftAssetType = 0
+	azure microsoftAssetType = 1
 )
 
 var (
 	_ providers.Instance           = (*Provider)(nil)
 	_ providers.PlatformIdentifier = (*Provider)(nil)
 )
 
-// New create a new Microsoft 365 provider
+// New create a new Microsoft provider
```

Comment:
I think we should update the description to include also ways how to call it for azure and ms365. The specific example is also outdated and we should update it